### PR TITLE
Docs for build tool

### PR DESCRIPTION
**Pull request by SamMarkowitz**

### CODE REVIEW

**Comment by meirwah** [comment]
@Bonczidai look at this.. see if it helps you to understand the new test framework...

**Comment by meirwah** [comment]
@Bonczidai did you find issues with docs (regarding test framework)

**Comment by Bonczidai** [comment]
no, I think it's explained pretty clear

maybe later when the testing mechanism will be "stable" we should add a section about how to actually write tests (environment, docker-in-docker, inputs like host, port etc.)

**Comment by meirwah** [comment]
@orius123 / @orius123  can we merge this?

**Comment by orius123** [comment]
No one got to review it yet.. I guess you meant @oritstone

File: app/markdowns/slang.md [hunk 1]
```diff
@@ -1476,11 +1476,65 @@ operation:
     - num: ID
 ```
 
+##CloudSlang Tests
+CloudSlang tests are written to test CloudSlang content and are run during the build process by the [CloudSlang Build Tool](#/docs#cloudslang-build-tool).
+
```
Comment (oritstone) [inline app/markdowns/slang.md]: Link to build-tool docs does not work

File: app/markdowns/slang.md [hunk 2]
```diff
@@ -1476,11 +1476,65 @@ operation:
...
+Wrapper flows are often used to set up an environment before the test runs and to clean up the environment after the test. They are also sometimes necessary for complex tests of a flow or operation's outputs.
+
+Wrapper flows are written in CloudSlang using the **.sl** extension and use the normal flow syntax. 
+
+###Test Suites
+Test suites are groups of tests that are only run if the build declares them as active. Test suites are often used to group tests that require a certain environment that may or may not be present in order to run. When the environment is present the suite can be activated and when it is not present the tests will not run.  
+
+Tests declare which test suites they are a part of, if any, using the `testSuites` property. 
+
+###Test Syntax
```
Comment (oritstone) [inline app/markdowns/slang.md]: Should be - Test case syntax

File: app/markdowns/slang.md [hunk 3]
```diff
@@ -1476,11 +1476,65 @@ operation:
...
+CloudSlang test files are written in YAML with the .yaml extension and contain one or more test cases.
```
Comment (oritstone) [inline app/markdowns/slang.md]: CloudSlang test case files are... with the .inputs.yaml extension ...

File: app/markdowns/slang.md [hunk 4]
```diff
@@ -1476,11 +1476,65 @@ operation:
...
+Property|Required|Value Type|Description
+---|---|---
+`inputs`|no|list of key:value pairs|inputs to pass to the flow or operation being tested
+`description`|no|string|description of test case
+`testFlowPath`|yes|string|qualified name of the flow, operation or wrapper flow to test
+`testSuites`|no|list|list of suites this test belongs to
+`outputs`|no|list of key:value pairs|expected output values of the flow, operation or wrapper flow being tested
+`result`|no|flow or operation result|expected result of the flow, operation or wrapper flow being tested
+`throwsException`|no|boolean|whether or not to expect an exception
+
```
Comment (oritstone) [inline app/markdowns/slang.md]: Missing: systemPropertiesFile

File: app/markdowns/slang.md [hunk 5]
```diff
@@ -1476,11 +1476,65 @@ operation:
...
+Wrapper flows are often used to set up an environment before the test runs and to clean up the environment after the test. They are also sometimes necessary for complex tests of a flow or operation's outputs.
+
+Wrapper flows are written in CloudSlang using the **.sl** extension and use the normal flow syntax. 
+
+###Test Suites
+Test suites are groups of tests that are only run if the build declares them as active. Test suites are often used to group tests that require a certain environment that may or may not be present in order to run. When the environment is present the suite can be activated and when it is not present the tests will not run.  
+
+Tests declare which test suites they are a part of, if any, using the `testSuites` property. 
+
```
Comment (oritstone) [inline app/markdowns/slang.md]: Missing - if no test suite is defined, the test case will always run

**Comment by SamMarkowitz** [comment]
@oritstone  Couldn't find an example with systemPropertiesFile. Can you check to see I got it right?

**Comment by SamMarkowitz** [comment]
@oritstone added your description

**Pull request merged**